### PR TITLE
Added '--json' option which allows to dump all stats.

### PR DESCRIPTION
**Pull request by mpeuster**
New option '--json <path>' allows to dump all collected statistics to the given JSON file at the end
of a run. This makes reusing/parsing the results with
other tools much easier.
Implemented for publish and subscribe.

### CODE REVIEW

File: beem/__init__.py [hunk 1]
```diff
@@ -44,6 +45,18 @@ def print_publish_stats(stats):
...
+def json_dump_stats(stats, path):
+    """
+    write the stats object to disk.
+    """
+    try:
+       with open(path, 'w') as f:  
+           json.dump(stats, f)
+       print("Wrote stats to: {}".format(path))
+    except IOError:
+        print "Couldn't dump JSON stats to: {}".format(path)
```
Comment (karlp) [inline beem/__init__.py]: you should at least use the function form here so there's a chacne of this working with py3.  (and just to be consistent with other uses)

File: beem/cmds/publish.py [hunk 2]
```diff
@@ -207,8 +210,13 @@ def run(options):
...
         beem.print_publish_stats(agg_stats)
+        if options.json is not None:
+            beem.json_dump_stats(agg_stats, options.json)
     else:
         agg_stats_set = [beem.aggregate_publish_stats(x) for x in stats_set]
         for x in agg_stats_set:
             x["time_total"] = time_end - time_start
         [beem.print_publish_stats(x) for x in agg_stats_set]
+        if options.json is not None:
+            [beem.json_dump_stats(x, options.json) for x in agg_stats_set]
```
Comment (karlp) [inline beem/cmds/publish.py]: this is only going to write the last file, it will just keep truncating it each time.  Is that really what you meant?

**Comment by mpeuster** [comment]
You are right. Overlooked this, was in a rush. I push a fix in some minutes.

**Comment by karlp** [comment]
merged via rebase in 15b11cf thanks.

**Pull request closed without merging**